core/simulations/helpers/createVaultWithCollateral.ts

```diff
@@ -160,18 +167,32 @@ const giveJoinContractAllowance = async (collateralConfig: CollateralConfig, amo
     await contract.approve(addressJoin, amountRaw);
 };
 
-const createVaultWithCollateral = async (collateralType: CollateralType, collateralOwned: BigNumber) => {
-    const [balanceSlot, languageFormat] = await determineBalanceSlot(collateralType);
-    const collateralConfig = getCollateralConfigByType(collateralType);
-
-    if (balanceSlot && languageFormat) {
-        await setCollateralInWallet(collateralConfig.symbol, collateralOwned);
-    } else {
-        // fallback to setting vat balance and withdrawing it
-        await setAndCheckCollateralInVat(collateralType, collateralOwned);
-        await checkAndWithdrawCollateralFromVat(collateralConfig, collateralOwned);
+const createProxiedVaultWithCollateral = async (collateralType: CollateralType, collateralOwned: BigNumber) => {
+    const { minUnitPrice, stabilityFeeRate } = await fetchVaultCollateralParameters(TEST_NETWORK, collateralType);
+    const collateralSymbol = getCollateralConfigByType(collateralType).symbol;
+    const drawnDebtExact = collateralOwned.multipliedBy(minUnitPrice).dividedBy(stabilityFeeRate);
+    const drawnDebt = roundDownToFirstSignificantDecimal(drawnDebtExact);
+    const proxyAddress = await createProxy(TEST_NETWORK, HARDHAT_PUBLIC_KEY);
+    await giveAllowanceToAddress(TEST_NETWORK, collateralSymbol, proxyAddress, collateralOwned);
+    await openVaultWithProxiedContractAndDrawDebt(
+        TEST_NETWORK,
+        proxyAddress,
+        collateralType,
+        collateralOwned,
+        drawnDebt
+    );
+    const registry = await getContract(TEST_NETWORK, 'CDP_REGISTRY');
+    const filter = registry.filters.NewCdpRegistered(null, proxyAddress, null);
+    const vaultOpenEvents = await registry.queryFilter(filter);
+    const vaultId = vaultOpenEvents[vaultOpenEvents.length - 1].args?.cdp;
```

Comment:
Same here. Wouldn't `vaultId` (`cdpi` in maker terms) be returned by the open method from the `CDP_REGISTRY.open` call?

In either way, please move the code to determine opened vault into `openVaultWithProxiedContractAndDrawDebt` so it will return this number, just like `createDefaultVaultWithCollateral` does it

Reply:
> Same here

as above.

Reply:
See my suggestion above